crates/cli/src/subcommands/dns.rs

```diff
@@ -1,97 +1,37 @@
-        Command::new("lookup")
-            .about("Resolves a domain to a database identity")
-            .arg(
-                Arg::new("domain")
-                    .required(true)
-                    .help("The name of the domain to lookup"),
-            )
-            .arg(
-                common_args::server()
-                    .help("The nickname, host name or URL of the server on which to look up the domain name"),
-            )
-            .after_help("Run `spacetime dns lookup --help` for more detailed information"),
-        Command::new("reverse-lookup")
-            .about("Returns the domains for the provided database identity")
-            .arg(
-                Arg::new("database-identity")
-                    .required(true)
-                    .help("The database identity you would like to find all of the known domains for"),
-            )
-            .arg(
-                common_args::server()
-                    .help("The nickname, host name or URL of the server on which to look up the database identity"),
-            )
-            .after_help("Run `spacetime dns reverse-lookup --help` for more detailed information.\n"),
-        Command::new("set-name")
-            .about("Sets the domain of the database")
-            .arg(
-                Arg::new("domain")
-                    .required(true)
-                    .help("The domain you would like to assign or create"),
-            )
-            .arg(
-                Arg::new("database-identity")
-                    .required(true)
-                    .help("The database identity to assign to the domain"),
-            )
-            .arg(common_args::server().help("The nickname, host name or URL of the server on which to set the name"))
-            .after_help("Run `spacetime dns set-name --help` for more detailed information.\n"),
-    ]
-}
-
-async fn exec_subcommand(config: Config, cmd: &str, args: &ArgMatches) -> Result<(), anyhow::Error> {
-    match cmd {
-        "register-tld" => exec_register_tld(config, args).await,
-        "lookup" => exec_dns_lookup(config, args).await,
-        "reverse-lookup" => exec_reverse_dns(config, args).await,
-        "set-name" => exec_set_name(config, args).await,
-        unknown => Err(anyhow::anyhow!("Invalid subcommand: {}", unknown)),
-    }
-}
-
-async fn exec_register_tld(mut config: Config, args: &ArgMatches) -> Result<(), anyhow::Error> {
-    let tld = args.get_one::<String>("tld").unwrap().clone();
+    let domain = args.get_one::<String>("new-name").unwrap();
+    let database_identity = args.get_one::<String>("database-identity").unwrap();
     let server = args.get_one::<String>("server").map(|s| s.as_ref());
+    let identity = decode_identity(&config)?;
+    let auth_header = get_auth_header(&config, false)?;
 
-    match spacetime_register_tld(&mut config, &tld, server).await? {
+    match spacetime_register_tld(&config, domain, server).await? {
```

Comment:
note: this new `exec` function is a combination of the previous `exec_register_tld` and `exec_set_name` functions, now all-in-one.